### PR TITLE
Video history

### PR DESCRIPTION
**Pull request by jeetbhalani15**
## Added Video History Functionality
- Added History Page layout.
- Added Video to history after watching using API Calls.
- Delete History using API.

### CODE REVIEW

File: Gravity-tube/src/Reducers/FetchVideoReducer.js [hunk 1]
```diff
@@ -8,6 +8,12 @@ function FetchVideoReducer(state,{type,payload}) {
...
         case "REMOVE_FROM_WATCH_LATER":
             return {...state, watchLater: payload}
+        case "ADD_TO_LIKED_VIDEOS":
+            return {...state, likedVideos: payload}
+        case "REMOVE_FROM_LIKED_VIDEOS":
+            return {...state, likedVideos: payload}
+        case "ADD_TO_HISTORY":
+            return {...state, history: payload}
             
             break;
```
Comment (sanket4120) [inline Gravity-tube/src/Reducers/FetchVideoReducer.js]: can remove break

File: Gravity-tube/src/Pages/SingleVideoPage/Component/VideoDetailPage.js [hunk 2]
```diff
@@ -67,18 +105,20 @@ function VideoDetailPage({}) {
                 >
                   {videoDataState.watchLater.find(
                     (item) => item._id === currentVideo._id
-                  ) ? (<>
-                    <MdOutlineAddTask color="#00D4C1" size={25} />
-                    <span style={{color:"#00D4C1"}}>Watch Later</span>
+                  ) ? (
+                    <>
+                      <MdOutlineAddTask color="#00D4C1" size={25} />
+                      <span style={{ color: "#00D4C1" }}>Watch Later</span>
```
Comment (sanket4120) [inline Gravity-tube/src/Pages/SingleVideoPage/Component/VideoDetailPage.js:111]: avoid the use of inline styles

File: Gravity-tube/src/Pages/VideoHistory/Components/VideoHistoryPage.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        <span className='watchlater-views'>
+         {videoDataState.history.length} Videos • 0 views • updated today</span>
+         <div className="tag">
+         <RiLock2Line color='grey' size={12}/><span className='tag-txt'>Private</span>
+         </div>
+         <hr/>
+        </div>
+        <div className="watch-later-videos">
+        <div className="watchlater-video-box">
+            {console.log(videoDataState.watchLater)}
```
Comment (sanket4120) [inline Gravity-tube/src/Pages/VideoHistory/Components/VideoHistoryPage.js]: remove console logs if not required

**Review by sanket4120 — APPROVED** [review]
Rest looks good

File: Gravity-tube/src/Contexts/Auth-context.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ const AuthProvider = ({ children }) => {
   const [authState, authDispatch] = useReducer(AuthReducer, { token: null });
   const getToken = localStorage.getItem("token");
   const getUser = JSON.parse(localStorage.getItem("user")); 
+  console.log(getUser)
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Auth-context.js:11]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 5]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
 };
 
 
+// ADD TO LIKED VIDEOS
+const likedVideoHandler = async (authState,video,videoDataDispatch) => {
+  if (authState.token) {
+    try {
+      console.log(video);
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 6]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
...
+      const res = await axios.post(
+        "/api/user/likes",
+        { video },
+        { headers: { authorization: authState.token } }
+      );
+      console.log(res);
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 7]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
...
+      videoDataDispatch({
+        type: "ADD_TO_LIKED_VIDEOS",
+        payload: res.data.likes,
+      });
+    } catch (error) {
+      alert(error);
+      console.log(error);
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 8]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
...
+    navigate("/login");
+  }
+};
+
+
+//  REMOVE FROM LIKE VIDEOS USING API
+const removeFromLikedVideos = async (authState,id,videoDataDispatch) => {
+  if (authState.token) {
+    try {
+      console.log(id);
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 9]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
...
+      const res = await axios.delete(
+        `/api/user/likes/${id}`,
+        { headers: { authorization: authState.token } }
+      );
+      console.log(res);
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 10]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
...
+      videoDataDispatch({
+        type: "REMOVE_FROM_LIKED_VIDEOS",
+        payload: res.data.likes,
+      });
+      console.log("removed")
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 11]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
...
+    } catch (error) {
+      alert(error);
+      console.log(error);
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 12]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
...
+    navigate("/login");
+  }
+};
+
+
+// ADD TO VIDEOS HISTORY
+const videoHistoryHandler = async (authState,video,videoDataDispatch) => {
+  if (authState.token) {
+    try {
+      console.log(video);
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 13]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
...
+      const res = await axios.post(
+        "/api/user/history",
+        { video },
+        { headers: { authorization: authState.token } }
+      );
+      console.log(res);
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Contexts/Videos-context.js [hunk 14]
```diff
@@ -80,9 +80,84 @@ const removeFromWatchLater = async (authState,id,videoDataDispatch) => {
...
+      videoDataDispatch({
+        type: "ADD_TO_HISTORY",
+        payload: res.data.history,
+      });
+    } catch (error) {
+      alert(error);
+      console.log(error);
```
Comment (Naveenchandar) [inline Gravity-tube/src/Contexts/Videos-context.js]: Please remove console.log

File: Gravity-tube/src/Pages/LikedVideos/Components/LikedVideosPage.js [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+        </span>
+        <div className="tag">
+          <RiLock2Line color="grey" size={12} />
+          <span className="tag-txt">Private</span>
+        </div>
+        <hr />
+      </div>
+      <div className="watch-later-videos">
+        <div className="watchlater-video-box">
+          {console.log(videoDataState.watchLater)}
```
Comment (Naveenchandar) [inline Gravity-tube/src/Pages/LikedVideos/Components/LikedVideosPage.js:28]: Please remove console.log

File: Gravity-tube/src/Pages/LikedVideos/Components/LikedVideosPage.js [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+          {videoDataState.likedVideos.map((video) => (
+              <>
+           <li className="list-num"></li> <HorizontalVideoCard video={video} />
```
Comment (Naveenchandar) [inline Gravity-tube/src/Pages/LikedVideos/Components/LikedVideosPage.js]: li should be used with ordered list or unordered list as parent element. Please change either parent element or li.

File: Gravity-tube/src/Pages/LikedVideos/Components/LikedVideosPage.js [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+          <RiLock2Line color="grey" size={12} />
+          <span className="tag-txt">Private</span>
+        </div>
+        <hr />
+      </div>
+      <div className="watch-later-videos">
+        <div className="watchlater-video-box">
+          {console.log(videoDataState.watchLater)}
+          {videoDataState.likedVideos.map((video) => (
+              <>
```
Comment (Naveenchandar) [inline Gravity-tube/src/Pages/LikedVideos/Components/LikedVideosPage.js:30]: No need of using fragments here. Please use key attribute while mapping data.

File: Gravity-tube/src/Pages/VideoHistory/Components/VideoHistoryPage.js [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+            
+            {videoDataState.history.reverse().map((video)=>(<HorizontalVideoCard video={video}/>))}
```
Comment (Naveenchandar) [inline Gravity-tube/src/Pages/VideoHistory/Components/VideoHistoryPage.js]: Please use key props while mapping data.

File: Gravity-tube/src/Pages/WatchLater/Components/HorizontalVideoCard.js [hunk 19]
```diff
@@ -40,6 +42,12 @@ function HorizontalVideoCard({video}) {
                   Remove from Watch later
                 </span>
               </div>
+              <div className="options">
+                  <RiDeleteBin6Line size={25} />
+                <span onClick={()=>removeFromLikedVideos(authState,video._id,videoDataDispatch)}>
+                  Remove from Liked Videos
+                </span>
+              </div>
               {/* <div className="options">
```
Comment (Naveenchandar) [inline Gravity-tube/src/Pages/WatchLater/Components/HorizontalVideoCard.js:45]: Please remove commented code if not required

**Pull request merged**